pkg/querier/worker/scheduler_processor.go

```diff
@@ -153,7 +155,16 @@ func (sp *schedulerProcessor) querierLoop(c schedulerpb.SchedulerForQuerier_Quer
 			}
 			logger := util_log.WithContext(ctx, sp.log)
 
-			sp.runRequest(ctx, logger, request.QueryID, request.FrontendAddress, request.StatsEnabled, request.HttpRequest)
+			switch r := request.Request.(type) {
+			case *schedulerpb.SchedulerToQuerier_HttpRequest:
+				sp.runHTTPRequest(ctx, logger, request.QueryID, request.FrontendAddress, request.StatsEnabled, r.HttpRequest)
+			case *schedulerpb.SchedulerToQuerier_QueryRequest:
+				sp.runQueryRequest(ctx, logger, request.QueryID, request.FrontendAddress, request.StatsEnabled, r.QueryRequest)
+			default:
+				// todo: how should we handle the error here?
```

Comment:
do we want to do something different here for now? or just remove the todo?

Reply:
I removed it.